public/app/features/explore/LiveLogs.tsx

```diff
@@ -119,6 +126,15 @@ class LiveLogs extends PureComponent<Props, State> {
     return rowsToRender;
   };
 
+  /**
+   * Check if row is fresh so we can apply special styling. This is bit naive and does not take into account rows
+   * which arrive out of order. Because loki datasource sends full data instead of deltas we need to compare the
+   * data and this is easier than doing some intersection of some uuid of each row (which we do not have now anyway)
+   */
+  isFresh = (row: LogRowModel): boolean => {
```

Comment:
@ryantxu @torkelo has any of you have a better idea here? Loki datasource knows which rows are new when it is appending the to it's buffer, but if we are going to throttle the events (we do in the PanelQueryRunner already) the information would be lost because Loki would mark which are new between every event, but we will pass an event only every Xms for example. I would assume @ryantxu answer would to put also the throttling inside the datasource, which would solve this (for the cost of some more complexity for the datasource) but then I would left it for some next PR.

Reply:
`row.timeEpochMs > this.state.lastTimestamp` seems like a good approach.

I think we can/should try other throttling options, but will be easier after other work is done (return Observables etc)